Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/ContactDiaryDayViewModel.kt

```diff
@@ -15,14 +16,15 @@ import org.joda.time.LocalDate
 
 class ContactDiaryDayViewModel @AssistedInject constructor(
     dispatcherProvider: DispatcherProvider,
+    systemInfoProvider: DefaultSystemInfoProvider,
     @Assisted selectedDay: String
 ) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
     private val displayedDay = MutableStateFlow(LocalDate.parse(selectedDay))
 
     val routeToScreen: SingleLiveEvent<ContactDiaryDayNavigationEvents> = SingleLiveEvent()
 
     val uiState = displayedDay.map { day ->
-        UIState(dayText = day.toFormattedDay())
+        UIState(dayText = day.toFormattedDay(systemInfoProvider))
```

Comment:
You can use this pattern here or create a variant of it to resolve the string with a context shortly before displaying it:

https://github.com/corona-warn-app/cwa-app-android/blob/349c7072282dcce3cd87bab1b1224146ba9aa87b/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/ui/LazyString.kt#L5-L7

Reply:
I'm not sure if I got your point here. You want to put a context into the ViewModel?

Reply:
Ideally we want to use the view's context to resolve the date. As close to the place where we display the date as possible.

* We could inject the application context into the viewmodel, but that's the application context again, same issue as previously.
* We could use assisted inject to make the fragment's context accessible from within the viewmodel but that's not a nice separation of concerns.
* We could change the `UIState.dayText` into a function that takes a context and returns a formatted day.
    * Could do that by just creating a lambda, e.g. `dayText: (Context) -> String`
    * Or if this is something that would be used else were too, created a new implementation of the interface `LazyString` (see other implementations in that file).

Reply:
Thank you for the detailed explanation! I decided to use the lambda because I was afraid of introducing side effects with `LazyString` like in #1958 and I think it does not make a big difference at this point. Also reverted all the SystemInfoProvider stuff (the cleanup of its current usages should be done in a separate PR I guess to not mix up too much in this one).